File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character,Integer> seens = new HashMap<>();


        for(int i = 0; i < s.length(); i++)
        {
            seens.put(s.charAt(i),seens.getOrDefault(s.charAt(i),0) + 1);
        }
        for(int i = 0; i < s.length(); i++)
        {
            if(seens.get(s.charAt(i)) == 1)
                return i;
        }

        return -1;
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:

        map = {}
        for i in range(len(s)):
            map[s[i]] = map.get(s[i],0) + 1

        for k,v in enumerate(s):
            if map[v] == 1:
                return k
        return -1

        